fulp_modules/features/antagonists/devil_affairs/devil_affairs.dm

```diff
@@ -0,0 +1,183 @@
+/datum/antagonist/traitor/devil_affairs
+	name = "\improper Devil Affairs Agent"
+	roundend_category = "devil affairs agents"
+	suicide_cry = "FOR MY SOUL!!"
+	job_rank = ROLE_DEVIL_AFFAIRS
+	preview_outfit = /datum/outfit/devil_affair_agent
+	should_give_codewords = FALSE
+
+	///List of all targets we have stolen thus far.
+	var/list/datum/mind/targets_stolen = list()
+
+/datum/antagonist/traitor/devil_affairs/on_gain()
+	. = ..()
+	ADD_TRAIT(owner.current, TRAIT_DEFIB_BLACKLISTED, REF(src))
+	RegisterSignal(owner.current, COMSIG_LIVING_REVIVE, .proc/on_revive)
+	RegisterSignal(owner.current, COMSIG_LIVING_DEATH, .proc/on_death)
+	uplink_handler.has_progression = FALSE
+	uplink_handler.has_objectives = FALSE
+	pick_employer()
+
+/datum/antagonist/traitor/devil_affairs/on_removal()
+	UnregisterSignal(owner.current, COMSIG_LIVING_DEATH)
+	UnregisterSignal(owner.current, COMSIG_LIVING_REVIVE)
+	REMOVE_TRAIT(owner.current, TRAIT_DEFIB_BLACKLISTED, REF(src))
+	return ..()
+
+/datum/antagonist/traitor/devil_affairs/apply_innate_effects(mob/living/mob_override)
+	. = ..()
+	var/mob/living/datum_owner = mob_override || owner.current
+	datum_owner.apply_status_effect(/datum/status_effect/agent_pinpointer/devil_affairs)
+
+/datum/antagonist/traitor/devil_affairs/remove_innate_effects(mob/living/mob_override)
+	var/mob/living/datum_owner = mob_override || owner.current
+	datum_owner.remove_status_effect(/datum/status_effect/agent_pinpointer/devil_affairs)
+	return ..()
+
+/// Affairs Agents can only roll Internal or External, rather than the normal factions.
+/datum/antagonist/traitor/devil_affairs/pick_employer(faction)
+	employer = ROLE_DEVIL_AFFAIRS
+	traitor_flavor = strings(DEVILAFFAIR_FLAVOR_FILE, employer, directory = "fulp_modules/strings")
+
+/datum/antagonist/traitor/devil_affairs/greet()
+	. = ..()
+	var/crime = pick(
+		"distribution of contraband",
+		"unauthorized erotic action on duty",
```

Comment:
I know it is just flavor text or something, but are you sure about using erotic reference or stuff idk.